Standalone IT with system property set improperly should fail

While trying to run the ITs against a standalone cluster, the system properties "-Dit.test=AccumuloClientIT -Dtest=foo -Daccumulo.it.properties=/path/to/accumulo-it.properties"
are required but if the path is wrong to the file or there are problems with the properties file, then the framework will default to Mini, and pass.  

It would be nice if these properties are set, most importantly the property file, that we do some validation and at the very least fail fast with a clear error ascertaining the problem.

Thanks for the comments.  After reading the recommended links I believe what I need to run is

man verify -Dit.test=AccumuloClientIT -Dtest=foo -Daccumulo.it.properties=/path/to/accumulo-it.properties

or maybe 

mvn verify -Dfailsafe.groups=org.apache.accumulo.test.categories.StandaloneCapableClusterTests     -Dit.test=AccumuloClientIT -Dtest=foo -Daccumulo.it.properties=/path/to/accumulo-it.properties

To get familiar with the mvn verify ITs I tried to run this off master:
mvn verify -Dfailsafe.groups=org.apache.accumulo.test.categories.MiniClusterOnlyTests

I thought I should see the MAC ITs complete before trying the standalone ITs, but I got this error:

[INFO] --- spotbugs-maven-plugin:3.1.7:spotbugs (spotbugs) @ accumulo-core ---
[INFO] Fork Value is true
     [java] The following errors occurred during analysis:
     [java]   Unable to locate method java.util.HashSet.forEach(Ljava/util/function/Consumer;)V
     [java]   Unable to locate method java.util.HashSet.removeIf(Ljava/util/function/Predicate;)Z
     [java]   Unable to locate method java.util.HashSet.stream()Ljava/util/stream/Stream;
     [java] Warnings generated: 1
     [java] Analysis errors: 3
[INFO] Done SpotBugs Analysis....
[INFO] 
[INFO] <<< spotbugs-maven-plugin:3.1.7:check (run-spotbugs) < :spotbugs @ accumulo-core <<<
[INFO] 
[INFO] 
[INFO] --- spotbugs-maven-plugin:3.1.7:check (run-spotbugs) @ accumulo-core ---
[INFO] BugInstance size is 1
[INFO] Error size is 3
[INFO] Total bugs: 1
[ERROR] Null passed for non-null parameter of ScannerOptions.fetchColumn(IteratorSetting$Column) in org.apache.accumulo.core.clientImpl.ScannerOptionsTest.testFetchNullColumn() [org.apache.accumulo.core.clientImpl.ScannerOptionsTest] At ScannerOptionsTest.java:[line 83] NP_NULL_PARAM_DEREF_ALL_TARGETS_DANGEROUS
[INFO] 

I got the same error with:
mvn verify -Dfailsafe.groups=org.apache.accumulo.test.categories.StandaloneCapableClusterTests     -Dit.test=AccumuloClientIT -Dtest=foo -Daccumulo.it.properties=/path/to/accumulo-it.properties

Please let me know what I am missing so these tests will complete.

Thanks.

I am getting a lot closer, but I have another question.  Basically, I am working in this class

AccumuloClusterPropertyConfiguration

and changing the appropriate log statements to RuntimeExceptions.  It works with the example

mvn verify -Dit.test=AccumuloClientIT -Dtest=foo -Daccumulo.it.properties=/path/to/accumulo-it.properties

with an error saying the file cannot be found instead of just switching to the MAC.

Then I wanted to see it actually work, so I tried the same line but with the correct path to the accumulo-it.properties that is in my uno directory and I got this error:

[ERROR]   AccumuloClientIT.testClose:146 expected:<CLIENT> but was:<SERVER>

Am I supposed to use a different accumulo-it.properties and somehow designate it as a client?  I am not sure how to do that and I don't see it addressed in the docs.  Any help would be appreciated.

Thanks.

I just realized the error did not print correctly because it used gt lt brackets like html.  With those removed the error looks like:

[ERROR] Failures: 
[ERROR]   AccumuloClientIT.testClose:146 expected:CLIENT but was:SERVER

Am I supposed to use a different accumulo-it.properties and somehow designate it as a client? I am not sure how to do that and I don't see it addressed in the docs. Any help would be appreciated.

Thanks.

--- a/test/src/main/java/org/apache/accumulo/harness/conf/AccumuloClusterPropertyConfiguration.java
+++ b/test/src/main/java/org/apache/accumulo/harness/conf/AccumuloClusterPropertyConfiguration.java
@@ -63,19 +63,19 @@ public abstract class AccumuloClusterPropertyConfiguration implements AccumuloCl
         try {
           reader = new FileReader(f);
         } catch (FileNotFoundException e) {
-          log.warn("Could not read properties from specified file: {}", propertyFile, e);
+          throw new RuntimeException("Could not read properties from specified file: " + propertyFile, e);
         }
 
         if (reader != null) {
           try {
             fileProperties.load(reader);
           } catch (IOException e) {
-            log.warn("Could not load properties from specified file: {}", propertyFile, e);
+            throw new RuntimeException("Could not load properties from specified file: " + propertyFile, e);
           } finally {
             try {
               reader.close();
             } catch (IOException e) {
-              log.warn("Could not close reader", e);
+              throw new RuntimeException("Could not close reader", e);
             }
           }
 
@@ -83,10 +83,10 @@ public abstract class AccumuloClusterPropertyConfiguration implements AccumuloCl
           clientConf = fileProperties.getProperty(ACCUMULO_CLUSTER_CLIENT_CONF_KEY);
         }
       } else {
-        log.debug("Property file ({}) is not a readable file", propertyFile);
+        throw new RuntimeException("Property file (" + propertyFile + ") is not a readable file");
       }
     } else {
-      log.debug("No properties file found in {}", ACCUMULO_IT_PROPERTIES_FILE);
+      throw new RuntimeException("No properties file found in " + ACCUMULO_IT_PROPERTIES_FILE);
     }
 
     if (clusterTypeValue == null) {

--- a/test/src/main/java/org/apache/accumulo/harness/conf/AccumuloClusterPropertyConfiguration.java
+++ b/test/src/main/java/org/apache/accumulo/harness/conf/AccumuloClusterPropertyConfiguration.java
@@ -57,36 +57,16 @@ public abstract class AccumuloClusterPropertyConfiguration implements AccumuloCl
     if (propertyFile != null) {
       // Check for properties provided in a file
       File f = new File(propertyFile);
-      if (f.exists() && f.isFile() && f.canRead()) {
-        Properties fileProperties = new Properties();
-        FileReader reader = null;
-        try {
-          reader = new FileReader(f);
-        } catch (FileNotFoundException e) {
-          throw new RuntimeException("Could not read properties from specified file: " + propertyFile, e);
-        }
-
-        if (reader != null) {
-          try {
-            fileProperties.load(reader);
-          } catch (IOException e) {
-            throw new RuntimeException("Could not load properties from specified file: " + propertyFile, e);
-          } finally {
-            try {
-              reader.close();
-            } catch (IOException e) {
-              throw new RuntimeException("Could not close reader", e);
-            }
-          }
-
-          clusterTypeValue = fileProperties.getProperty(ACCUMULO_CLUSTER_TYPE_KEY);
-          clientConf = fileProperties.getProperty(ACCUMULO_CLUSTER_CLIENT_CONF_KEY);
-        }
-      } else {
-        throw new RuntimeException("Property file (" + propertyFile + ") is not a readable file");
+      Properties fileProperties = new Properties();
+      try (FileReader reader = new FileReader(f)) {
+        fileProperties.load(reader);
+        clusterTypeValue = fileProperties.getProperty(ACCUMULO_CLUSTER_TYPE_KEY);
+        clientConf = fileProperties.getProperty(ACCUMULO_CLUSTER_CLIENT_CONF_KEY);
+      } catch (IOException e) {
+        throw new RuntimeException("Could not read properties from file: " + propertyFile, e);
       }
     } else {
-      throw new RuntimeException("No properties file found in " + ACCUMULO_IT_PROPERTIES_FILE);
+      log.debug("No properties file found in {}", ACCUMULO_IT_PROPERTIES_FILE);
     }
 
     if (clusterTypeValue == null) {

